tests/integration_tests/utils.py

```diff
@@ -236,64 +236,120 @@ def number_feature(normalization=None, **kwargs):
     return feature
 
 
-def category_feature(**kwargs):
+def category_feature(output_feature=False, **kwargs):
+    if DECODER in kwargs:
+        output_feature = True
     feature = {
         "type": "category",
         "name": "category_" + random_string(),
-        ENCODER: {"type": "dense", "vocab_size": 10, "embedding_size": 5},
     }
+    if output_feature:
+        feature.update(
+            {
+                DECODER: {"type": "classifier", "vocab_size": 10},
+            }
+        )
+    else:
+        feature.update(
+            {
+                ENCODER: {"type": "dense", "vocab_size": 10, "embedding_size": 5},
+            }
+        )
     recursive_update(feature, kwargs)
     feature[COLUMN] = feature[NAME]
     feature[PROC_COLUMN] = compute_feature_hash(feature)
     return feature
 
 
-def text_feature(**kwargs):
+def text_feature(output_feature=False, **kwargs):
+    if DECODER in kwargs:
+        output_feature = True
     feature = {
         "name": "text_" + random_string(),
         "type": "text",
-        ENCODER: {
-            "type": "parallel_cnn",
-            "vocab_size": 5,
-            "min_len": 7,
-            "max_len": 7,
-            "embedding_size": 8,
-            "state_size": 8,
-        },
     }
+    if output_feature:
+        feature.update(
+            {
+                DECODER: {"type": "generator", "vocab_size": 5, "max_len": 7},
+            }
+        )
+    else:
+        feature.update(
+            {
+                ENCODER: {
+                    "type": "parallel_cnn",
+                    "vocab_size": 5,
+                    "min_len": 7,
+                    "max_len": 7,
+                    "embedding_size": 8,
+                    "state_size": 8,
+                },
+            }
+        )
     recursive_update(feature, kwargs)
     feature[COLUMN] = feature[NAME]
     feature[PROC_COLUMN] = compute_feature_hash(feature)
     return feature
 
 
-def set_feature(**kwargs):
+def set_feature(output_feature=False, **kwargs):
+    if DECODER in kwargs:
+        output_feature = True
     feature = {
         "type": "set",
         "name": "set_" + random_string(),
-        ENCODER: {"type": "embed", "vocab_size": 10, "max_len": 5, "embedding_size": 5},
     }
+    if output_feature:
+        feature.update(
+            {
+                DECODER: {"type": "classifier", "vocab_size": 10, "max_len": 5},
+            }
+        )
+    else:
+        feature.update(
+            {
+                ENCODER: {"type": "embed", "vocab_size": 10, "max_len": 5, "embedding_size": 5},
+            }
+        )
     recursive_update(feature, kwargs)
     feature[COLUMN] = feature[NAME]
     feature[PROC_COLUMN] = compute_feature_hash(feature)
     return feature
 
 
-def sequence_feature(**kwargs):
+def sequence_feature(output_feature=False, **kwargs):
```

Comment:
Just a couple comments here. I thought of this option for refactoring these feature utils during my encoder/decoder PR however realized that this would require going back through all the Ludwig tests and making sure all output features have this flag which takes forever. Is there a plan in a followup PR to go through and refactor all these tests? Also since you've added this output feature flag to some of the feature utils, should we add them to all that have an encoder in the util config definition (i.e. image_feature)?

Reply:
I added the `output_feature` kwarg, but ended up not really needing to use it anywhere.  

The conditional:
```
    if DECODER in kwargs:
        output_feature = True
```
was sufficient to handle all of our tests.  I thought about removing the `output_feature` kwarg and simplifying to `output_feature = DECODER in kwargs`, but decided not to because:
1.  Specifying output_feature=True might be useful for future test code.
2. `def sequence_feature(output_feature=False` makes it obvious that this could generate either an input or output feature.  I prefer having the explicit option, though that is just a style preference.

Reply:
Gotcha, ya I'm ok leaving it in and just having tests be refactored as needed going forward / new tests using it if necessary. Do you think we should add that kw arg in for the other feature gen utils then?

Reply:
The simple ones like binary don't really need it, nor image/audio/timeseries which don't have output features yet

Reply:
Makes sense. LGTM then!